### PR TITLE
Revisions

### PR DESCRIPTION
**Pull request by mjpost**
This makes changes to how we handle revisions:

- an explanation is required for each revision (old ones have been backfilled)
- this explanation is displayed with a tooltip next to each revision
- updated language: the label for first "revision" now reads "original"

I haven't pushed up the changes to the XML yet, to give people a chance to comment.

image

### CODE REVIEW

**Comment by knmnyn** [comment]
This is a good idea, but can it be a hidden division that get expanded instead of a tooltip?
The tooltip is good for short descriptions, but sometimes a revision will have more than can comfortably fit in a tooltip.

Secondly it would facilitate accessibility in printing, cutting and pasting.  

The form should explicitly highlight that this explanation will be made publicly visible and should show an example.

**Comment by mjpost** [comment]
These are good ideas. I'm also considering being more descriptive in the XML. Instead of

    <revision id="2" href="P18-1001v2">Explanation</revision>

I'd do

```
<revision id="2">
    <url>P18-1001v2</url>
    <explanation>Explanation</explanation>
</revision>
```

The reason is for parallelism with other `<url>` tags on main papers. It would be nice to consolidate on having `<url>` always for links (which either get expanded as absolute or relative).

I'm also wondering if we should use `<version>` instead of `<revision>`. The weird thing is that revision ID 2 is actually the *first* revision, but the second *version*. This is a nitpicky thing but perhaps worth straightening out.

**Comment by knmnyn** [comment]
I agree with @mjpost on the first issue here.  Note that we have both
<revision> and <erratum> tags currently, so I would propose we do the same
for both.
I don't think the new management has had to deal with retractions but they
would probably need a <explanation> tag as well.

Because we have both <erratum> and <revision>, the distinction between ids
is a bit difficult.  So while I think revision might make sense to merge
with ID and version, the <erratum> tag would not be as a parallel vehicle
(which would not be a totally bad thing in my opinion).

Cheers,

Min

--
Min-Yen KAN (Dr) :: Associate Professor :: National University of Singapore
:: NUS School of Computing, AS6 05-12, 13 Computing Drive
Singapore 117417 :: +65 6516 1885(DID) :: +65 6779 4580 (Fax) ::
kanmy@comp.nus.edu.sg (E) :: www.comp.nus.edu.sg/~kanmy (W)

**Comment by mjpost** [comment]
This too is ready for review. I ended up modifying the `<revision>` tag only slightly. Revisions now take an explanation which is recorded in the XML, but is not currently displayed on the pages, because my visual design skills are not good enough for me to do this efficiently. I didn't touch erratum.

File: hugo/static/google8ff8feffd5c8f6ac.html [hunk 1]
```diff
@@ -0,0 +1 @@
+google-site-verification: google8ff8feffd5c8f6ac.html
```
Comment (knmnyn) [inline hugo/static/google8ff8feffd5c8f6ac.html]: This file doesn't seem to be related to the issue of revisions.  Just FYI.  Not a blocking issue.

Comment (mjpost) [reply hugo/static/google8ff8feffd5c8f6ac.html]: This is for verifying the Google search console for my account (@mbollmann has one, too)

**Comment by mjpost** [comment]
Thanks! You're right on logic + content.
